### PR TITLE
WT-11765 Fix predictable replays not playing well with ops.throttling

### PR DESCRIPTION
**Pull request by ershov**
Summary of changes:

* Disable throttling when predictable replays are on.
* Disable throttling in certain configs where it can't possibly be useful:
  * `CONFIG.replay`
  * `CONFIG.coverage`
  * `CONFIG.endian`

### CODE REVIEW

**Comment by ershov** [comment]
I'm going to test the change with a few hundreds of runs in Evergreen.

File: test/format/format_config.c [hunk 1]
```diff
@@ -496,6 +496,19 @@ config_run(void)
           REALLOC_MAX_TABLES);
     }
 
+    if (GV(RUNS_PREDICTABLE_REPLAY)) {
+        /*
+         * Predictable replays can get extremely slow with throttling.
+         *
+         * TODO: Investigate reasons for that.
```
Comment (etienneptl) [inline test/format/format_config.c:503]: Can we have a fixme comment with a ticket id instead here of the todo here please?

File: test/format/format_config.c [hunk 2]
```diff
@@ -496,6 +496,19 @@ config_run(void)
...
+         */
+        if (GV(OPS_THROTTLE)) {
+            if (config_explicit(NULL, "ops.throttle"))
```
Comment (etienneptl) [inline test/format/format_config.c:506]: Should we check that it is explicit **and** on? It doesnt really matter in the end but i believe we would see that warning message even though we explicitly disabled throttling?

Comment (ddanderson) [reply test/format/format_config.c:506]: @etienneptl I believe `GV(OPS_THROTTLE)` in the line before will only be true when throttling is on.

File: test/format/format_config.c (hunk 1 above)
Comment (ddanderson) [reply test/format/format_config.c:503]: If there's already a new ticket, I agree.  If not, I'm okay with dropping the comment and not having a ticket.  I think we know already: that the interaction is slow, that we don't understand completely why, but it's likely to be a little complex, and that it doesn't matter that much.  Without throttling, we'd do a slightly less rigorous testing of predictable replay, but PR isn't a typical way of running WT.

**Review by ddanderson — APPROVED** [review]
I'm good with the change - except for the TODO comment that @etienneptl tagged.  Either add a ticket number to the TODO, or remove the TODO.

File: test/format/format_config.c (hunk 1 above)
Comment (ershov) [reply test/format/format_config.c]: Added ticket number: WT-11782

File: test/format/format_config.c (hunk 2 above)
Comment (etienneptl) [reply test/format/format_config.c:507]: That's right, thank you, @ddanderson!

**Review by etienneptl — APPROVED** [review]
LGTM!

**Pull request merged**